52.md

```diff
@@ -26,11 +26,15 @@ The `.content` of these events should be a detailed description of the calendar
 
 The list of tags are as follows:
 * `d` (required) universally unique identifier (UUID). Generated by the client creating the calendar event.
-* `title` (required) title of the calendar event
+* `type` (required) type of the event, list of [_onsite_, _online_]
 * `start` (required) inclusive start date in ISO 8601 format (YYYY-MM-DD). Must be less than `end`, if it exists.
 * `end` (optional) exclusive end date in ISO 8601 format (YYYY-MM-DD). If omitted, the calendar event ends on the same date as `start`.
-* `location` (optional, repeated) location of the calendar event. e.g. address, GPS coordinates, meeting room name, link to video call
+* `address` (required if `type` is _onsite_) address of the calendar event following ISO 19160:2015 and splitting the elements in the array; the country is specified in ISO 3166
+* `place` (optional, permitted only if `type` is _onsite_) name of the place of the calendar event. e.g. univeristy, building, business, public space
```

Comment:
I have the same opinion.

Reply:
The current `location` currently can contain... everything, so in the perspective of a smooth migration having a different name help clients to manage the old events vs the new ones

Reply:
I don't think there's anything wrong with `location` being able to contain anything. In fact, whether or not the spec says so, it can still contain anything -- including free-form addresses. Existing calendar events already put addresses in `location`.

Think about clients that don't want to incur the additional complexity of address formatting. They may just want to use `location` for everything.

The valuable part really seems like `address` is structured to the ISO spec. Interactive clients can make good use of that with address formatting and validation. But I don't think it should be strictly necessary.

Reply:
Flockstr is currently the main calendar client (right?) and generates this:

```
    [
      "location",
      "Happy Basset Brewing Co, 6044 SW 29th St, Topeka, KS 66614, USA",
      "Happy Basset Brewing Co",
      "6044 SW 29th St, Topeka, KS 66614, USA"
    ],
    [
      "address",
      "Happy Basset Brewing Co, 6044 SW 29th St, Topeka, KS 66614, USA",
      "Happy Basset Brewing Co",
      "6044 SW 29th St, Topeka, KS 66614, USA"
    ],
```

https://njump.me/naddr1qqyr2e3cxcer2epeqy8hwumn8ghj7mn0wd68ytnddaksygqf9wdhvvhdxtc9qj232zqexyfp0htz4xtcfnva0zjepll0829vkgpsgqqq0jesxlc44m

I see a lot of wrong things here :)